src/adaptivespatialsampling.jl

```diff
@@ -0,0 +1,79 @@
+"""
+    AdaptiveSpatialSampling
+
+...
+
+**numpoints**, an Integer (def. 50), specifying the number of points to use.
+
+**α**, an AbstractFloat (def. 1.0), specifying ...
+"""
+Base.@kwdef mutable struct AdaptiveSpatialSampling{T <: Integer} <: BONRefiner
+    numpoints::T = 50
+    function AdaptiveSpatialSampling(numpoints)
+        if numpoints < one(numpoints)
+            throw(ArgumentError("You cannot have an AdaptiveSpatialSampling with fewer than one point"))
+        end
+        return new{typeof(numpoints)}(numpoints)
+    end
+end
+
+function _generate!(coords::Vector{CartesianIndex}, pool::Vector{CartesianIndex}, sampler::AdaptiveSpatialSampling, uncertainty::Matrix{T}) where {T<:AbstractFloat}
+
+    # Distance matrix (inlined)
+    d = zeros(Float64, Int((sampler.numpoints * (sampler.numpoints - 1)) / 2))
```

Comment:
Should we have a `numpoints(sampler)` method?

Reply:
I've been going back and forth on this... it can be `numpoints` or `length`, but I like the predictability of using `numpoints` as a field. I don't really care (in the sense of: I'm fine with either decision)